instrumentation/httpclient5/src/test/java/brave/httpclient5/ITTracingH2AsyncClientBuilder.java

```diff
@@ -0,0 +1,216 @@
+package brave.httpclient5;
+
+import brave.test.http.ITHttpAsyncClient;
+import brave.test.util.AssertableCallback;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BiConsumer;
+import okhttp3.Protocol;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.apache.hc.client5.http.async.methods.SimpleHttpRequest;
+import org.apache.hc.client5.http.async.methods.SimpleHttpRequests;
+import org.apache.hc.client5.http.async.methods.SimpleHttpResponse;
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.client5.http.impl.async.H2AsyncClientBuilder;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClientBuilder;
+import org.apache.hc.core5.concurrent.FutureCallback;
+import org.apache.hc.core5.http.ContentType;
+import org.junit.Before;
+import org.junit.Test;
+
+import static brave.Span.Kind.CLIENT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class ITTracingH2AsyncClientBuilder extends ITHttpAsyncClient<CloseableHttpAsyncClient> {
+
+  static void invoke(CloseableHttpAsyncClient client, SimpleHttpRequest req) throws IOException {
+    Future<SimpleHttpResponse> future = client.execute(req, null);
+    blockOnFuture(future);
+  }
+
+  static <V> V blockOnFuture(Future<V> future) throws IOException {
+    try {
+      return future.get(3, TimeUnit.SECONDS);
```

Comment:
just be lazy and throw new IOException(e) for wide catch Exception as it is too much distraction in the test file and outcome would be similar as long as surefire is set to include stacktrace (which it is)